src/Ledger/TokenAlgebra/ValueSet.lagda

```diff
@@ -0,0 +1,249 @@
+open import Axiom.Set using (Theory)
+
+module Ledger.TokenAlgebra.ValueSet (PolicyId : Type)(th : Theory {lzero}) where
+
+open import Algebra                       using (CommutativeMonoid ; Commutative ; Op₂ ; Monoid)
+open import Algebra.Morphism              using ( IsMonoidMorphism ; IsSemigroupMorphism )
+
+open import Axiom.Set.Map th              using (_ˢ ; module Lookupᵐ ; singletonᵐ ; ❴_❵ᵐ)
+                                          renaming ( Map to _⇀_ )
+open import Axiom.Set.Rel th              using ( dom )
+
+open import Data.Nat                      using (ℕ ; _≤_ ; _+_)
+open import Data.Nat.Properties           using (+-0-commutativeMonoid)
+
+open import Ledger.Prelude                using ( DecEq )
+open import Ledger.TokenAlgebra.Base      using ( TokenAlgebra ; Quantity)
+
+open import Prelude                       using ( _,_ ; _×_ ; module Equivalence; _∘_ ; _≗_ ; id )
+open import Prelude                       using ( _≡_ ; refl ; 0ℓ; trans ; sym)
+
+open import Relation.Binary               using (IsEquivalence )
+
+open Theory th                            using (_∈_ ; spec-∈ ; singleton ; ∈-singleton)
+                                          renaming (map to mapˢ ; Set to Set' )
+open import Relation.Binary.PropositionalEquality.Core using ( module ≡-Reasoning )
+import Relation.Binary.Core  as stdlib
+
+\end{code}
+
+\subsubsection{Derived types}
+
+(See Fig 3 of the
+\href{https://github.com/input-output-hk/cardano-ledger/releases/latest/download/mary-ledger.pdf}%
+{Mary ledger specification}.)
+
+\begin{itemize}
+\item \AgdaBound{AssetName} is a byte string used to distinguish different assets with the same \AgdaBound{PolicyId}.
+\item \AgdaBound{AssetId} is a product type consisting of a \AgdaBound{PolicyId} and an \AgdaBound{AssetName}.
+\item \AgdaBound{AdaId} is the Id for the asset Ada.
+\item \AgdaBound{Quantity} is the type of amounts of assets.
+\end{itemize}
+
+In the formal ledger specification \AgdaBound{AssetId} is sometimes viewed as a direct sum type,
+the inhabitants of which belong to either \AgdaBound{AdaIdType} or the product
+\AgdaBound{PolicyId}~\AgdaBound{×}~\AgdaBound{AssetName}; if we were adhering to that point of view,
+then we would have defined
+\AgdaBound{AssetId}
+  = \AgdaBound{AdaIdType}~\AgdaBound{⊎}~(\AgdaBound{PolicyId}~\AgdaBound{×}~\AgdaBound{AssetName}).
+
+Finally, we define a record type with a single inhabitant with which we may wish to
+represent the type of Ada (rather than viewing Ada as just another asset).
+
+\begin{code}
+record AdaIdType : Type where
+  instance constructor AdaId
+\end{code}
+
+
+\subsection{Definition of the value monoid}
+
+An inhabitant of `Value` is a map denoting a finite collection of quantities of assets.
+
+\begin{code}
+open TokenAlgebra
+open CommutativeMonoid renaming (_∙_ to _⋆_) hiding (refl ; sym ; trans)
+open Algebra
+
+
+Value-TokenAlgebra :  (assetNameType  : Type                                       )
```

Comment:
```suggestion
Value-TokenAlgebra :  (AssetName  : Type                                       )
```
Adding the suffix `Type` to a type is bad style. Think of `n : Nat` vs. `n : NatType`. I've also seen this in a bunch of other places in this PR.

Reply:
Yes, I went back and forth with this one... I settled on `assetNameType` because I thought that, in this instance especially, it would help to emphasize that this is a type and not an asset name; i.e., it's not an inhabitant of the asset name type, it *is* the asset name type!

Reply:
Btw, `TransactionStructure` has a similar suffix.  Is this also bad style? ...or not *as* bad? 😄